jib-core/src/main/java/com/google/cloud/tools/jib/registry/credentials/DockerConfigCredentialRetriever.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.registry.credentials;
+
+import com.google.cloud.tools.jib.http.Authorization;
+import com.google.cloud.tools.jib.http.Authorizations;
+import com.google.cloud.tools.jib.json.JsonTemplateMapper;
+import com.google.cloud.tools.jib.registry.credentials.json.DockerConfigTemplate;
+import com.google.common.annotations.VisibleForTesting;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import javax.annotation.Nullable;
+
+/**
+ * Retrieves registry credentials from the Docker config.
+ *
+ * <p>The credentials are searched in the following order (stopping when credentials are found):
+ *
+ * <ol>
+ *   <li>If there is an {@code auth} defined for a registry.
+ *   <li>Using the {@code credsStore} credential helper, if available.
+ *   <li>Using the credential helper from {@code credHelpers}, if available.
+ * </ol>
+ *
+ * @see <a
+ *     href="https://docs.docker.com/engine/reference/commandline/login/">https://docs.docker.com/engine/reference/commandline/login/</a>
+ */
+public class DockerConfigCredentialRetriever {
+
+  private static final Path DOCKER_CONFIG_FILE =
+      Paths.get(System.getProperty("user.home")).resolve(".docker").resolve("config.json");
```

Comment:
is this true for windows/mac too?

Reply:
Yes, on windows, the `user.home` would give `%USERPROFILE%`, whereas on the unix systems, it gives `$HOME`.

https://docs.docker.com/engine/reference/commandline/login/#privileged-user-requirement states the location of the config file.